libs/remix-ui/editor/src/lib/providers/codeActionProvider.ts

```diff
@@ -34,24 +34,9 @@ export class RemixCodeActionProvider implements monaco.languages.CodeActionProvi
           if (nodeAtPosition.parameters && !Array.isArray(nodeAtPosition.parameters) && Array.isArray(nodeAtPosition.parameters.parameters)) {
             const paramNodes = nodeAtPosition.parameters.parameters
             // If method has parameters
-            if (paramNodes.length) {
-              // Get last function parameter node
-              const lastParamNode = paramNodes[paramNodes.length - 1]
-              const location = await this.props.plugin.call('codeParser', 'getLineColumnOfNode', lastParamNode)
-              // Get end location of last function parameter, it returns end column of parameter name
-              const lastParamEndLoc = location.end
-              const lineContent = model.getLineContent(lastParamEndLoc.line + 1)
-              if (fix.id === 5 && lineContent.includes(' view ')) {
-                msg = lineContent.replace('view', 'pure')
-              } else
-                msg = lineContent.substring(0, lastParamEndLoc.column + 2) + fix.message + lineContent.substring(lastParamEndLoc.column + 1, lineContent.length)
-              fix.range = {
-                startLineNumber: lastParamEndLoc.line + 1,
-                endLineNumber: lastParamEndLoc.line + 1,
-                startColumn: 0,
-                endColumn: error.startColumn + msg.length
-              } 
-            } else {
+            if (paramNodes.length)
+              msg = await this.fnWithParamsQFMsg(model, paramNodes, fix, error, true) 
```

Comment:
Is it possible to reconsider a more descriptive function name? fnWithParamsQFMsg can be better I think.

Reply:
done